addons/common/functions/fnc_progressBar.sqf

```diff
@@ -22,22 +24,34 @@
  * Public: Yes
  */
 
-params ["_totalTime", "_args", "_onFinish", "_onFail", ["_localizedTitle", ""], ["_condition", {true}], ["_exceptions", []]];
+params ["_totalTime", "_args", "_onFinish", "_onFail", ["_localizedTitle", ""], ["_condition", {true}], ["_exceptions", []], ["_dialog", true]];
 
 private _player = ACE_player;
 
 //Open Dialog and set the title
 closeDialog 0;
-createDialog QGVAR(ProgressBar_Dialog);
+if (_dialog) then {
+    createDialog QGVAR(ProgressBar_Dialog);
+} else {
+    QGVAR(progressBarDisplay) cutRsc [QGVAR(ProgressBar_Display), "PLAIN"];
+};
 
 private _display = uiNamespace getVariable QGVAR(dlgProgress);
 
 // Ensure CBA keybindings are hooked into the display
 _display call (uiNamespace getVariable "CBA_events_fnc_initDisplayCurator");
 
 // Hide cursor by using custom transparent cursor
-private _map = _display displayCtrl 101;
-_map ctrlMapCursor ["", QGVAR(blank)];
+if (_dialog) then {
+    private _map = _display displayCtrl 101;
+    _map ctrlMapCursor ["", QGVAR(blank)];
+} else { // Add key handler for ESC to cancel
+    [DIK_ESCAPE, [false, false, false], {
+        QGVAR(progressBarDisplay) cutText ["", "PLAIN"];
+        [QGVAR(progressBarKeyHandler), "keydown"] call CBA_fnc_removeKeyHandler;
+        true
+    }, "keydown", QGVAR(progressBarKeyHandler)] call CBA_fnc_addKeyHandler;
```

Comment:
They key handler is only removed if ESC is pressed. Could this be problematic if the display is closed by other means? The key handler would still be active.

Reply:
Good catch.